### PR TITLE
REGR: assert_series_equal defaulting to check_exact=True for Index

### LINKED ISSUE #57067: BUG: testing.assert_series_equal: inferred check_exact should not be passed down to index check
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> a = pd.Series(np.zeros(6, dtype=int), [0, 0.2, 0.4, 0.6, 0.8, 1])
>>> b = pd.Series(np.zeros(6, dtype=int), np.linspace(0, 1, 6))  # 0.6000000000000001
>>> pd.testing.assert_series_equal(a, b)
```

### Issue Description

In pandas 2.2.0, the above passes if the series' values dtype is float and fails if the series' values dtype is int. In pandas 2.1.4, it passes in both cases.

The difference seems to be caused by a change in the default for the check_exact in `assert_series_equal`:

> Changed in version 2.2.0: Defaults to True for integer dtypes if none of check_exact, rtol and atol are specified.

What I think is happening is that, in pandas 2.1.4, `assert_series_equal` was passing down its own default `check_exact=False` to `assert_index_equal` - even if the default for the latter is `check_exact=True`. In 2.2.0, `assert_series_equal` infers `check_exact=True` from the values dtype, and then passes it down to `assert_index_equal`.

### Expected Behavior

In my opinion both pandas 2.2.0 and 2.1.4 are wrong here. If the user doesn't explicitly state `check_exact`, it should default to two potentially different values for the values and the index, which means False for float values, True for int values, and True for the index regardless of dtype (as it already is for `assert_index_equal`).

### Installed Versions

<details>
commit                : f538741432edf55c6b9fb5d0d496d2dd1d7c2457
python                : 3.12.1.final.0
python-bits           : 64
OS                    : Linux
OS-release            : 6.5.0-14-generic
Version               : #14~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov 20 18:15:30 UTC 2
machine               : x86_64
processor             : x86_64
byteorder             : little
LC_ALL                : None
LANG                  : en_GB.UTF-8
LOCALE                : en_GB.UTF-8

pandas                : 2.2.0
numpy                 : 1.26.3
</details>

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #57067 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

File: pandas/_testing/asserters.py [hunk 1]
```diff
@@ -902,6 +902,7 @@ def assert_series_equal(
     >>> tm.assert_series_equal(a, b)
     """
     __tracebackhide__ = True
+    check_exact_index = False if check_exact is lib.no_default else check_exact
```
Comment (rhshadrach) [inline pandas/_testing/asserters.py:905]: Don't we want to default to check_exact=True for integers and false for floats?

Comment (phofl) [reply pandas/_testing/asserters.py:905]: I wasn't a fan of doing the breaking change in a minor release, that's why I reverted back to 2.1.x behaviour for Index. We can introduce the more intelligent behavior for 3.0 if someone wants to do it

Comment (rhshadrach) [reply pandas/_testing/asserters.py:905]: Opened #57386 to track.

**Pull request merged**